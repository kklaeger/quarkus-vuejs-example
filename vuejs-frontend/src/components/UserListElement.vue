<template>
  <div class="columns is-mobile">
    <div class="column is-3-mobile is-2-tablet is-1-desktop is-2-widescreen">
      <b-icon pack="fas" icon="user" size="is-medium"> </b-icon>
    </div>
    <div class="column name-column">{{ user.name }}</div>
    <div class="column is-3-mobile is-2-tablet is-1-desktop is-2-widescreen">
      <b-button rounded type="is-text"
        ><b-icon pack="fas" icon="trash" size="is-medium"> </b-icon
      ></b-button>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from "vue-property-decorator";
import { User } from "@/models/User";

@Component({
  components: {}
})
export default class UserListElement extends Vue {
  @Prop({ required: true }) user!: User;
}
</script>

<style scoped lang="scss">
.name-column {
  text-align: left;
  padding-top: 15px;
  padding-bottom: 15px;
}
</style>
