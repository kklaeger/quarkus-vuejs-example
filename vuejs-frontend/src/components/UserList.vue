<template>
  <div>
    <div v-if="this.users.length == 0">
      No users available :)
    </div>
    <div v-else>
      <ul>
        <UserListElement
          v-for="user in this.users"
          :key="user.id"
          :user="user"
        />
      </ul>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from "vue-property-decorator";
import UserListElement from "@/components/UserListElement.vue";
import { User } from "@/models/User";

@Component({
  components: { UserListElement }
})
export default class UserList extends Vue {
  @Prop({ required: true }) users!: User[];
}
</script>

<style scoped lang="scss"></style>
