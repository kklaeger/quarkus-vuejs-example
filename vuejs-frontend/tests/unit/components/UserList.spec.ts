import { shallowMount, createLocalVue } from "@vue/test-utils";
import Buefy from "buefy";
import UserList from "@/components/UserList.vue";
import UserListElement from "@/components/UserListElement.vue";
import { User } from "@/models/User";

const localVue = createLocalVue();
localVue.use(Buefy);

describe(UserList.name + " component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render the user list for empty user list and match snapshot.", () => {
    const users: User[] = [];

    const wrapper = shallowMount(UserList, {
      localVue,
      propsData: { users }
    });

    expect(wrapper.findAllComponents(UserListElement)).toHaveLength(0);
    expect(wrapper.text()).toMatch("No users available :)");
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the user list for one user and match snapshot.", () => {
    const users: User[] = [{ id: 1, name: "user1" }];

    const wrapper = shallowMount(UserList, {
      localVue,
      propsData: { users }
    });

    expect(wrapper.findAllComponents(UserListElement)).toHaveLength(1);
    expect(
      wrapper
        .findAllComponents(UserListElement)
        .at(0)
        .props().user
    ).toEqual({ id: 1, name: "user1" });
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the user list for two users and match snapshot.", () => {
    const users: User[] = [
      { id: 1, name: "user1" },
      { id: 2, name: "user2" }
    ];

    const wrapper = shallowMount(UserList, {
      localVue,
      propsData: { users }
    });

    expect(wrapper.findAllComponents(UserListElement)).toHaveLength(2);
    expect(
      wrapper
        .findAllComponents(UserListElement)
        .at(0)
        .props().user
    ).toEqual({ id: 1, name: "user1" });
    expect(
      wrapper
        .findAllComponents(UserListElement)
        .at(1)
        .props().user
    ).toEqual({ id: 2, name: "user2" });
    expect(wrapper).toMatchSnapshot();
  });
});
