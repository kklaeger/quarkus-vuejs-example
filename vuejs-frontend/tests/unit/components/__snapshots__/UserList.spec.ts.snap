// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList component should render the user list for empty user list and match snapshot. 1`] = `
<div>
  <div>
    No users available :)
  </div>
</div>
`;

exports[`UserList component should render the user list for one user and match snapshot. 1`] = `
<div>
  <div>
    <ul>
      <userlistelement-stub user="[object Object]"></userlistelement-stub>
    </ul>
  </div>
</div>
`;

exports[`UserList component should render the user list for two users and match snapshot. 1`] = `
<div>
  <div>
    <ul>
      <userlistelement-stub user="[object Object]"></userlistelement-stub>
      <userlistelement-stub user="[object Object]"></userlistelement-stub>
    </ul>
  </div>
</div>
`;
